### PR TITLE
Error handling for impl PoolReserve in pallet-pools

### LINKED ISSUE #646: Error handling for impl PoolReserve  in pallet-pools
Currently, we will face an `ArithmeticError::Underflow` error in case we withdraw more from a pool than possible.

The impl of the `PoolReserve` trait should instead handle errors with something like `NotEnoughFunds`.

### PR DESCRIPTION
**Pull request by stanly-johnson**
resolves #646

### CODE REVIEW

File: pallets/pools/src/lib.rs [hunk 1]
```diff
@@ -521,6 +521,8 @@ pub mod pallet {
 		NoSolutionAvailable,
 		/// One of the runtime-level pool parameter bounds was violated
 		PoolParameterBoundViolated,
+		/// Not enough funds in pool
+		NotEnoughFundsInPool,
```
Comment (NunoAlexandre) [inline pallets/pools/src/lib.rs:525]:
```suggestion
		InsufficientReserveFunds,
```

**Comment by branan** [comment]
I wonder if it makes more sense to use one of the built-in [`TokenError`](https://docs.substrate.io/rustdocs/latest/sp_runtime/enum.TokenError.html) variants instead of creating our own error?

**Comment by mustermeiszer** [comment]
@branan the `TokenError` enum is not very descriptive in my opinion. The only roughly matching variant would be `TokenError::NoFunds`, that translates into  "Funds are unavailable" as the error message. I would keep it as an PoolErorr so one knows the reason directly.

**Review by mustermeiszer — CHANGES_REQUESTED** [review]
Thanks for the contribution @stanly-johnson! 

Referring to what @branan said, I think it makes sense to use the `TokenError::NoFunds` variant here instead of using a costume error message.

**Review by branan — APPROVED** [review]
Looks good to me. Once tests have run I believe we can hit the button.

Thanks for your contribution!

**Comment by NunoAlexandre** [comment]
@stanly-johnson 👏

**Pull request merged**